chain/jsonrpc-primitives/src/errors.rs

```diff
@@ -15,11 +15,28 @@ pub struct RpcParseError(pub String);
 #[serde(deny_unknown_fields)]
 pub struct RpcError {
     pub code: i64,
+    #[serde(flatten)]
+    pub error_struct: Option<RpcErrorKind>,
```

Comment:
Let's move it one line above, and also mark all the other fields as deprecated. Consider using [`#[deprecated(...)]` attribute](https://doc.rust-lang.org/edition-guide/rust-2018/the-compiler/an-attribute-for-deprecation.html) or just doc comments.